controllers/v1/cloud-services/files.js

```diff
@@ -79,110 +77,148 @@ module.exports = class Files {
+      } catch (error) {
+        return reject({
+          status:
+            error.status || httpStatusCode["internal_server_error"].status,
+
+          message:
+            error.message || httpStatusCode["internal_server_error"].message,
+
+          errorObject: error,
+        });
+      }
+    });
+  }
+
+  /**
+   * @api {post} /kendra/api/v1/cloud-services/files/getDownloadableUrl
+   * Get downloadable URL.
+   * @apiVersion 1.0.0
+   * @apiGroup Gcp
+   * @apiHeader {String} X-authenticated-user-token Authenticity token
+   * @apiParamExample {json} Request:
+   * {
+   *     "filePaths": []
+   * }
+   * @apiSampleRequest /kendra/api/v1/cloud-services/files/getDownloadableUrl
+   * @apiUse successBody
+   * @apiUse errorBody
+   * @apiParamExample {json} Response:
+   * {
+   *  "status": 200,
+   *  "message": "Url's generated successfully",
+   *  "result": [{
+   *  "filePath": "5e1c28a050452374e1cf9841/e97b5582-471c-4649-8401-3cc4249359bb/cdv_photo_117.jpg",
+   *  "url": "https://storage.googleapis.com/download/storage/v1/b/sl-dev-storage/o/5e1c28a050452374e1cf9841%2Fe97b5582-471c-4649-8401-3cc4249359bb%2Fcdv_photo_117.jpg?generation=1579240054787924&alt=media"
+   * }]
+   */
+
+  /**
+   * Get Downloadable URL from cloud service.
+   * @method
+   * @name getDownloadableUrl
+   * @param  {Request}  req  request body.
+   * @returns {JSON} Response with status and message.
+   */
+
+  async getDownloadableUrl(req) {
+    return new Promise(async (resolve, reject) => {
+      try {
+        let downloadableUrl = await filesHelpers.getDownloadableUrl(
+          req.body.filePaths
+        );
+
+        return resolve(downloadableUrl);
+      } catch (error) {
+        console.log(error);
+        return reject({
+          status:
+            error.status || httpStatusCode["internal_server_error"].status,
+
+          message:
+            error.message || httpStatusCode["internal_server_error"].message,
+
+          errorObject: error,
+        });
+      }
+    });
+  }
+
+  /**
+      * upload the file to the cloud .
       * @method
-      * @name preSignedUrls
+      * @name upload
       * @param  {Request}  req  request body.
-      * @param  {Array}  req.body.fileNames - list of file names
-      * @param  {String}  req.body.bucket - name of the bucket  
-      * @returns {JSON} Response with status and message.
-    */
+      * @returns {JSON} Response with status .
+      * @apiParamExample {json} Response:
+     * {
+        "status": 200
+       }
+     */
 
-   async preSignedUrls(req) {
+  async upload(req) {
     return new Promise(async (resolve, reject) => {
-
-        try {
-
-            let signedUrl =
-            await filesHelpers.preSignedUrls(
-                 req.body.request,
-                 req.body.ref,
-                 req.userDetails ? req.userDetails.userId : ""
+      try {
+
+        const filename = path.basename(req.query.file);
+          if(filename === req.files.file.name){
```

Comment:
if its binary then don't validate just upload and if its form-data then upload.
and its not necessary that you get form data with the file key only. First check if any key is present and then read data from that key. @praveenKDass

Reply:
Resolved, Added condition for binary